### DISCUSSION TITLE
关于排除文件的机制

### DISCUSSION THREAD

**Discussion by KawasumiMaii** [xmake-io/xmake#1770] [Q&A] [answered]
在xmake.lua文件中，当目录层级为如下的情况
├── src
│   └── subSrc
│       ├── main.cpp
│       └── test.cpp
└── xmake.lua
add_files("src/**/*.cpp|src/subSrc/test.cpp")
指定的文件并未被排除。。

**Comment by waruqi** [comment, marked as answer]
排除路径相对于前面 `**`之前的 src 目录

```lua
add_files("src/**/*.cpp|subSrc/test.cpp")
```

**Reply by KawasumiMaii** [reply]
谢谢！
